src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -91,7 +92,12 @@ public static SamInputResource of(final File file) {
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
     public static SamInputResource of(final Path path) {
-        return new SamInputResource(new PathInputResource(path));
+
+        if (Files.isRegularFile(path) &&  Files.exists(path)) {
+            return new SamInputResource(new PathInputResource(path));
+        } else {
+                return of(path.toFile());
```

Comment:
Indentation is funny

Reply:
I had originally implemented it as 
```
return of(new FileInputStream(path))
```
But that requires a try-catch block to guard against non-existant paths.
perhaps in the presence of the `Files.exists(path)` clause this is a non-issue and I can swallow the exception and return a null (which should never ever happen)?

happy to change it or not...as people prefer.